src/snap-points.ts

```diff
@@ -110,37 +115,164 @@ export const createSnapshotForNetwork = async (network: Network) => {
-          event,
-          ownerData,
-          market
+        const ownerData = eventsObject[ownerKey] || {
+          active: [],
+          closed: [],
+        };
+        const correspondingItemIndex = acc.newOpen.findIndex((item) =>
+          item.id.eq(event.id)
         );
-        if (updatedData) eventsObject[ownerKey] = updatedData;
-
-        break;
+        if (correspondingItemIndex >= 0) {
+          const correspondingItem = acc.newOpen[correspondingItemIndex];
+          acc.newOpen.splice(correspondingItemIndex, 1);
+          acc.newOpenClosed.push([correspondingItem, event]);
+          return acc;
+        }
+        const correspondingItemIndexPreviousData = ownerData.active.findIndex(
+          (item) => item.event.id.eq(event.id)
+        );
+        if (correspondingItemIndexPreviousData >= 0) {
+          const correspondingItem =
+            ownerData.active[correspondingItemIndexPreviousData];
+          acc.newClosed.push([correspondingItem, event]);
+          return acc;
+        }
+        acc.newOpenClosed.push([null, event]);
+        return acc;
       }
+      return acc;
+    },
+    { newOpen: [], newClosed: [], newOpenClosed: [] }
+  );
 
-      case InvariantEventNames.RemovePositionEvent: {
-        const event = parseEvent(decodedEvent) as RemovePositionEvent;
-        const ownerKey = event.owner.toString();
-        const ownerData = eventsObject[ownerKey] || { active: [], closed: [] };
-        const updatedData = processRemovePositionEvent(event, ownerData);
-        if (updatedData) eventsObject[ownerKey] = updatedData;
-        break;
+  const stillOpen: IActive[] = [];
+
+  Object.values(eventsObject).forEach((positions) =>
+    positions.active.forEach((activeEntry) => {
+      const hasBeenClosed = newClosed.some(
+        (newClosedEntry) => newClosedEntry[0].event.id === activeEntry.event.id
+      );
+      if (!hasBeenClosed) {
+        stillOpen.push(activeEntry);
       }
-      default:
-        break;
+    })
+  );
+
+  const poolsWithTicks: IPoolAndTicks[] = await Promise.all(
+    PROMOTED_POOLS.map(async (pool) => {
+      const ticksUsed = Array.from(
+        new Set([
+          ...stillOpen.flatMap((entry) =>
+            entry.event.pool.toString() === pool.toString()
+              ? [entry.event.lowerTick, entry.event.upperTick]
+              : []
+          ),
+          ...newOpen.flatMap((entry) =>
+            entry.pool.toString() === pool.toString()
+              ? [entry.lowerTick, entry.upperTick]
+              : []
+          ),
+        ])
+      );
+
+      const [poolStructure, ticks] = await Promise.all([
+        market.getPoolByAddress(pool),
+        Promise.all(ticksUsed.map((tick) => market.getTickByPool(pool, tick))),
+      ]);
+
+      return { pool, poolStructure: poolStructure, ticks };
+    })
+  );
+
+  const currentTimestamp = getTimestampInSeconds();
+
+  const convertedLastSnapTimestamp = new BN(lastSnapTimestamp);
```

Comment:
If we are saving the value as number it means that the domain till 2^53 is sufficient for us. There is no need to use it as BN in the code. You should change all timestamp and point references to `number`